safe/gui/tools/batch/batch_dialog.py

```diff
@@ -520,59 +521,32 @@ def run_task(self, task_item, status_item, count=0, index=''):
                 # Usually after analysis is done, the impact layer
                 # become the active layer. <--- WRONG
                 # noinspection PyUnresolvedReferences
-                impact_layer = self.dock.impact_function.impact
-
-                # Load impact layer into QGIS
-                qgis_layer = read_impact_layer(impact_layer)
-                impact_layer_source = qgis_layer.source()
-                # reg.instance().addMapLayer(qgis_layer, addToLegend=False)
-
-                # identify impact layer in map canvas from impact layer source
-                leg_impact_layer = self.identify_layer(impact_layer_source)
-                self.legend_impact_layer_is_removed = False
-                if leg_impact_layer.type() == QgsMapLayer.VectorLayer:
-                    leg_impact_layer.layerDeleted.connect(self.check_removed)
-                elif leg_impact_layer.type() == QgsMapLayer.RasterLayer:
-                    leg_impact_layer.destroyed.connect(self.check_removed)
-                # clone impact layer in Layer Panel
-                if qgis_layer.type() == QgsMapLayer.VectorLayer:
-                    cloned_layer = QgsVectorLayer(
-                        leg_impact_layer.source(),
-                        leg_impact_layer.name(),
-                        leg_impact_layer.providerType())
-                elif qgis_layer.type() == QgsMapLayer.RasterLayer:
-                    cloned_layer = QgsRasterLayer(
-                        leg_impact_layer.source(),
-                        leg_impact_layer.name(),
-                        leg_impact_layer.providerType())
-                else:
-                    raise Exception('layer source is failed to be recognized')
-                # remove original layer created from dock, sometimes this
-                # code is failed leaving original file in layer panel
-                reg.instance().removeMapLayer(leg_impact_layer)
-                # turn off layer visibility just in case layer removing
-                # is not working so that it won't be shown in the next scenario
+                impact_function = self.dock.impact_function
+                try:
+                    impact_layer = impact_function._exposure_impacted
```

Comment:
* You shouldn't use private members from the IF. You should always use public members.
* And you should use `impact_function.impact`, it will give the more detailed impacted layer (the exposure impacted or the aggregate_hazard_impacted, or at least the aggregation_impacted). So you can remove the try/except.